.github/workflows/test.yml

```diff
@@ -0,0 +1,106 @@
+name: Tests
```

Comment:
Why is this a separate workflow? Would it be better if we ran all the tests in each environment like the other flow and maybe just merge them together? Or maybe there's a better way to partition the workflows?